### PR TITLE
New Diagnostics tool

### PR DESCRIPTION
**Pull request by gingerwizard**
### Changelog category
- New Feature

### Changelog entry
New single binary based diagnostics tool.

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
> [tools/clickhouse-diagnostics/testdata/logs/var/logs/clickhouse-server.log.gz](https://github.com/ClickHouse/ClickHouse/pull/36705/files#diff-5785673a226a9aa79229765e914b7f6052864ea920daef75c8ad1225e2a735b7)

Is it needed?

File: tools/clickhouse-diagnostics/CONTRIBUTION.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+1. Identify current version
+
+```bash
+git tag
+```
+
+2. Create a new tag with a higher version number e.g.
+
+```bash
+git tag -a v1.0.0-beta.3 -m "A summary comment"
```
Comment (alexey-milovidov) [inline tools/clickhouse-diagnostics/CONTRIBUTION.md:67]: This becomes misleading.

File: tools/clickhouse-diagnostics/CONTRIBUTION.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+`git push origin v1.0.0-beta.3`
+
+4. Ensure you have a GITHUB_TOKEN set with permissions to the repository. Create a token [here](https://github.com/settings/tokens/new) with all repo scope permissions. Ensure this is set as an environment variable i.e.
+
+`export GITHUB_TOKEN=blah`
+
+5. Run a release
+
+```bash
+make release
```
Comment (alexey-milovidov) [inline tools/clickhouse-diagnostics/CONTRIBUTION.md:83]: But there is no makefile.

File: tools/clickhouse-diagnostics/CONTRIBUTION.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
+# Contribution
+
+We keep things simple. Execute all commands in the root folder.
```
Comment (alexey-milovidov) [inline tools/clickhouse-diagnostics/CONTRIBUTION.md:3]: root folder - "this folder"

**Comment by alexey-milovidov** [comment]
Let's also correct this: https://s3.amazonaws.com/clickhouse-test-reports/36705/f8dfe1906f9d19bef4855de0f51df2455dd318f9/style_check__actions_/style_output.txt

File: tools/clickhouse-diagnostics/CONTRIBUTION.md (hunk 2 above)
Comment (gingerwizard) [reply tools/clickhouse-diagnostics/CONTRIBUTION.md:83]: will add.

**Comment by gingerwizard** [comment]
> Let's also correct this: https://s3.amazonaws.com/clickhouse-test-reports/36705/f8dfe1906f9d19bef4855de0f51df2455dd318f9/style_check__actions_/style_output.txt

Thanks go-lint doesnt check the yaml.

> [tools/clickhouse-diagnostics/testdata/logs/var/logs/clickhouse-server.log.gz](https://github.com/ClickHouse/ClickHouse/pull/36705/files#diff-5785673a226a9aa79229765e914b7f6052864ea920daef75c8ad1225e2a735b7)
> 
> Is it needed?
> 

Its part of the tests to confirm gz logs are copied. I removed the file and replaced with a dummy file with an indicator as such i.e. its not just a text file to minimize space.

**Comment by alexey-milovidov** [comment]
One minor thing: https://s3.amazonaws.com/clickhouse-test-reports/36705/1c0e258c8bbb083bbfa751d9dc87ebefd2633d82/style_check__actions_/style_output.txt

**Comment by gingerwizard** [comment]
How to run this lint check locally over just a sub folder? I see build executes

```docker run -u $(id -u ${USER}):$(id -g ${USER}) --cap-add=SYS_PTRACE --volume=/home/ubuntu/actions-runner/_work/ClickHouse/ClickHouse:/ClickHouse --volume=/home/ubuntu/actions-runner/_work/_temp/style_check:/test_output clickhouse/style-test:latest```

Is there a simpler way to do locally?

**Pull request merged**